app/models/category_list.rb

```diff
@@ -64,17 +64,18 @@ def find_relevant_topics
 
     @all_topics = Topic.where(id: category_featured_topics.map(&:topic_id)).includes(:shared_draft, :category)
 
+    @all_topics = @all_topics.includes(:tags).where(tags: { name: @options[:tag] }) if @options[:tag].present?
```

Comment:
```suggestion
    @all_topics = @all_topics.joins(:tags).where(tags: { name: @options[:tag] }) if @options[:tag].present?
```

I think we want to do a join here instead of eager loading?